pkgs/applications/editors/neovim/wrapper.nix

```diff
@@ -100,6 +143,8 @@ let
       + optionalString (configure != {}) ''
         echo "Generating remote plugin manifest"
         export NVIM_RPLUGIN_MANIFEST=$out/rplugin.vim
+        ${optionalString withRuby "export GEM_HOME=${rubyEnv}/${rubyEnv.ruby.gemPath}"}
+        ${optionalString (binPath != "") "export PATH=$PATH:${binPath}"}
```

Comment:
@Infinisil I did set GEM_HOME here, I forgot `PATH` but I added it here now in https://github.com/NixOS/nixpkgs/pull/88110/commits/159895f6793c315656a78b523ecdeb4f52de3c00 (not the latest commit you reviewed).

The next commit -  https://github.com/NixOS/nixpkgs/pull/88110/commits/845a1371a0bc6abfcbf1c43c01d2f37bd0050c54, implements what you described:

> wrap the binary with those, call it for determining rplugin.vim, then call makeWrapper on the original unwrapped binary once again ...

And I'm not sure which is better. The nix variable's names are longer :hear_no_evil: and IMO it's a maybe a tiny bit harder to understand what's going on when there's two `makeWrapper` calls? I added comments which thoroughly explain how everything works.